File: pset6/mario/mario.py
from cs50 import get_int

"""
    // Autor: Luisángel Martín Marcia Palma
    // PSET 6 mario.py
"""

"""The function that creates the piramid"""
def piramid (height):
    for i in range (0, height, 1):
        for j in range(0, (height - (i+1)), 1):
            print(' ', end='')
        for k in range(0, i+2, 1):
            print('#', end='')
        print('')
    return


"""The function that helps to verify the input from the user"""
def getHeight():
    height = get_int("Height: ")

    while height < 0 or height > 23:
        height = get_int("Height: ")

    return int(height)


"""Main function"""
def main ():
    piramid(getHeight())
    return

if __name__ == ("__main__"):
    main()

File: pset1/mario/mario.c
#include<stdio.h>
#include<cs50.h>

// Autor: Luisángel Martín Marcia Palma
// PSET 1 mario.c

// Llamando función
int piramid (int cant);

// Función principal
int main (void)
{
    int height = 0;

    do
    {
        height = get_int("Height:");
        // Condición que impide que el usuario me digite valores negativos y mayores a 23 los cuales sugiere el pset
    }while (height < 0 || height > 23);

    piramid(height);
    return 0;
}
// Función piramid
int piramid (int column)
{
        int m = column;
        for (int i = 0; i < column; i++)
        {
            // Ciclo para que digite los espacios necesarios para formar el espacio en blanco de la piramide
            for (int j = 0; j < (column- (i + 1)); j++)
            {
                printf (" ");
            }
            // For secundario
            // Ciclo para que empiece a dibujar la piramide con el numeral
            for (int k = 0; k <= i + 1; k++)
            {
                printf ("#");
            }
            printf ("\n");
        }
    return 0;
}